### ISSUE TITLE
PopupMenuButton - tooltip not read by Android TalkBack (a11y issue)

### ISSUE THREAD

**Issue by benjaminwg2017** [flutter/flutter#171939]
### Steps to reproduce

On a Pixel 3A phone, Android version 12. 

1. Enable the TalkBack screen reader on a physical Android device or emulator.
2. Create a new Flutter project or use the provided DartPad link (https://dartpad.dev/?id=01e816ca464cef2658923fee5bd6b515)
3. Run the minimal sample code provided below.
4. Once the app is running, navigate the UI using TalkBack by swiping right.
5. When the accessibility focus lands on the PopupMenuButton (the three-dot icon) in the AppBar, listen carefully to the announcement made by TalkBack.

### Expected results

When the PopupMenuButton is focused, TalkBack should announce the text provided in the tooltip property, followed by the role of the widget.
For the sample code provided, the expected announcement should be: "show menu, Button".

### Actual results

The actual announcement is: "Button".

### Code sample

https://dartpad.dev/?id=01e816ca464cef2658923fee5bd6b515

### Flutter Doctor output

n/a

**Comment by darshankawar** [comment]
@benjaminwg2017 Thanks for the report. I was unable to use or repro with the given code, but I used [this](https://github.com/flutter/flutter/issues/147043#issue-2252703513) similar issue's code sample, added `tooltip` to it as below:

```
PopupMenuButton<String>(
            tooltip: 'this is test tooltip',
```

And with talkback ON, it read the tooltip properly as below:

First time when it gets the focus : `button, this is test tooltip, detected icon more button, double tap to activate`
Next time when it gets the focus : `detected, icon more button this is test tooltip, double tap to activate.`

I verified with latest master version : `3.33.0-1.0.pre.905`

Please see if above verification helps or not.

**Comment by benjaminwg2017** [comment]
I was running with version hash `7762b4cf32ac9e38282dd9b9d2e191b78bf0a289`, on a physical Android pixel 3A device, with Android version 12. 

And with talkback ON, it reads only:

`button, double tap to activate`

**Comment by darshankawar** [comment]
> used [this](https://github.com/flutter/flutter/issues/147043#issue-2252703513) similar issue's code sample,

@benjaminwg2017 Thanks for the update. Even with this code sample you get the observed behavior ?

**Comment by benjaminwg2017** [comment]
Yes, with the sample code in the linked issue edited as you suggest by adding a `tooltip` property, the behavior is buggy as I described on my pixel 3A physical device.

**Comment by darshankawar** [comment]
Thanks for the update @benjaminwg2017 
My colleague verified the same on his Pixel 6a (Android 16) and below are his findings:

- On latest master (3.33.0-1.0.pre.947) - it reads the tooltip properly as `Open menu Flutter SDK, button, double tap to activate` after passing:
```
PopupMenuButton<String>(
            // Semantics for the button itself. TalkBack will read this.
            tooltip:
                'Open menu Flutter SDK', // This shows on long press and is read by TalkBack.
```

But on latest stable, it does seem to ignore the tooltip, indicating the behavior is working as expected in master. Please try the same at your end.

**Comment by benjaminwg2017** [comment]
I ran the sample locally on master (3.33.0-1.0.pre.929); Could you let me know which commit I should sync to for (3.33.0-1.0.pre.947) so I could try running the sample code on it?

**Comment by darshankawar** [comment]
> I was running with version hash `7762b4cf32ac9e38282dd9b9d2e191b78bf0a289`

@benjaminwg2017 Is this pointing to `3.33.0-1.0.pre.929` ?

> Could you let me know which commit I should sync to for (3.33.0-1.0.pre.947) so I could try running the sample code on it?

I think if it is not working at your end on `3.33.0-1.0.pre.929`, it probably won't in `3.33.0-1.0.pre.947`, so looks like it is specific to the device you are using ? I am not sure about the exact commit, but just upgrading to latest `flutter upgrade` should give you the tip of the branch.

**Comment by benjaminwg2017** [comment]
Sorry for the back and forth... 

The issue seemed to have been resolved. When I was experiencing the issue, this is what I got when running `flutter --version`:

```
Flutter 3.33.0-1.0.pre.929 • channel master • git@github.com:gardenofwine/flutter.git
Framework • revision dcf22d1650 (4 days ago) • 2025-07-11 10:36:16 -0700
Engine • hash 3f6b25ab78a655e9cbbec4aeaad65ee2d5dfa59f (revision dcf22d1650) (3 days ago) • 2025-07-11 17:36:16.000Z
Tools • Dart 3.9.0 (build 3.9.0-327.0.dev) • DevTools 2.48.0
```

After executing the upgrade as you suggeted, I got:

```
Flutter 3.33.0-1.0.pre-971 • channel master • git@github.com:gardenofwine/flutter.git
Framework • revision 6e1ebc7a13 (3 hours ago) • 2025-07-15 04:29:34 -0700
Engine • hash c84613b4f6dd2848bda6452e01648f6c143e6ef7 (revision 6e1ebc7a13) (3 hours ago) • 2025-07-15 11:29:34.000Z
Tools • Dart 3.9.0 (build 3.9.0-333.0.dev) • DevTools 2.48.0
```

And the issue wouldn't reproduce anymore on my device. I'll be closing this issue then, but I am curious as to what could have changed between `3.33.0-1.0.pre.929` and `3.33.0-1.0.pre.971` to affect this bug.
